### PR TITLE
Add push guards to only process relevant review requests

### PR DESCRIPTION
**Pull request by gmiroshnykov**

### CODE REVIEW

File: rbbz/models.py [hunk 1]
```diff
@@ -77,6 +77,14 @@ def publish_review_request(user, review_request_draft, **kwargs):
 
 
 def publish_review(user, review, **kwargs):
+    # skip review requests that were not pushed
+    if str(review.review_request.extra_data.get('p2rb', False)) == "False":
```
Comment (smacleod) [inline rbbz/models.py]: You're defaulting to the value of |False|, and then checking if equal to the string "False", which will be False, so you won't return here in the case where p2rb isn't set.

Comment (gmiroshnykov) [inline rbbz/models.py]: The whole left hand side is wrapped in `str()`, so it should be fine.

Comment (smacleod) [inline rbbz/models.py]: Indeed. Woops!

File: rbbz/models.py [hunk 2]
```diff
@@ -91,6 +99,14 @@ def publish_review(user, review, **kwargs):
 
 
 def publish_reply(user, reply, **kwargs):
+    # skip review requests that were not pushed
```
Comment (mikeconley) [inline rbbz/models.py]: We're repeating ourselves a lot here. I wonder if it'd be worth having a decorator for these functions that does this stuff for us?

Comment (gmiroshnykov) [inline rbbz/models.py]: In those three cases, we're dealing with `extra_data` either in `review_request_draft`, `review.review_request` or `reply.review_request`.
I'm not sure how I can concisely hide this behind a decorator, but it's possible that I simply don't know Python that well :)
What would you recommend?

Comment (mikeconley) [inline rbbz/models.py]: Hm, yeah, a decorator would be awkward. Maybe just a separate function instead to do the checks for "p2rb" and "squashed". That's really the only repetition that I'm concerned with DRYing up.

File: rbbz/models.py [hunk 3]
```diff
@@ -69,29 +71,49 @@ def publish_review_request(user, review_request_draft, **kwargs):
...
 def publish_review(user, review, **kwargs):
-    bug_id = int(review.review_request.get_bug_list()[0])
+    review_request = review.review_request
+
+    # skip review requests that were not pushed
+    if str(review_request.extra_data.get('p2rb', False)) == "False":
+        return
+
+    # skip child review requests
+    if str(review_request.extra_data.get('squashed', False)) == "False":
```
Comment (mikeconley) [inline rbbz/models.py]: Also, I think this needs to be "p2rb.is_squashed", and not "squashed".

File: rbbz/models.py (hunk 2 above)
Comment (gmiroshnykov) [inline rbbz/models.py]: Done

File: rbbz/models.py (hunk 3 above)
Comment (gmiroshnykov) [inline rbbz/models.py]: Done

**Comment by mikeconley** [comment]
LGTM. Thanks George!

**Pull request merged**